lib/tapjoy/ldap/api/user.rb

```diff
@@ -0,0 +1,112 @@
+require 'memoist'
+module Tapjoy
+  module LDAP
+    module API
+      module User
+        class << self
+          extend Memoist
+          def create(fname, lname, type, group)
+            # Properly capitalize names
+            fname, lname = [fname, lname].map(&:titleize)
+
+            Tapjoy::LDAP::client.add(
+              distinguished_name(fname, lname, type),
+              ldap_attr(fname, lname, type, group)
+            )
+          end
+
+          def destroy(username, type)
+            Tapjoy::LDAP::client.delete(
+              distinguished_name(*name_of_user(username), type)
+            )
+          end
+
+          def index
+            Tapjoy::LDAP::client.search('*', filter(uid: '*'))
+          end
+
+          def show(username)
+            Tapjoy::LDAP::client.search('*', filter(uid: username))
+          end
+
+          private
+
+          # Filter users for #show and #index
+          def filter(uid: '*')
+            Net::LDAP::Filter.eq('uid', uid)
+          end
+
+          # Given a username, return First and Last names
+          def name_of_user(username)
+            username.split('.').map(&:titleize)
+          end
+          memoize :name_of_user
```

Comment:
Just double checking this: Since all of these are class methods, I'm assuming any call made to the api would be considered a brand new invocation of the gem and not have any constant process where these values will be cached form request to request correct?